File: Gramatica1.py
def gramatica1(cadena):
    
    #verificación que sea o y 1

    if not all(c in "01" for c in cadena):
        return False 
    
    #verifica capicúa 

    return cadena == cadena[::-1]

with open("G1.txt", "r") as archivo:
    for linea in archivo:
        cadena = linea.strip() 
        if gramatica1(cadena):
            print(f"{cadena}: acepta")
        else:
            print(f"{cadena}: No acepta")

File: Gramatica1.c
#include <stdio.h>
#include <string.h>

int gramatica1(char *cadena){
    int len = strlen(cadena);

    //verificar que cumpla la condición de 0 y 1

    for (int i =0; i<len; i++){
        if (cadena[i] != '0' && cadena[i] != '1')
            return 0;
    }

    for (int i = 0; i < len / 2; i++){


        if (cadena [i] != cadena[len -i - 1])
            return 0;

    }
    return 1;

}

int main() {
 char cadena[100];
    printf("Ingrese una cadena: ");
    scanf("%s", cadena);


    if (gramatica1(cadena)) {
        printf("Acepta\n");
    } else {
        printf("No acepta.\n");

        return 0;
    }

}